src/gg.regression.unity.bots/Tests/Scripts/RGOverlay/RGSegmentEntry.test.cs

```diff
@@ -0,0 +1,68 @@
+﻿using NUnit.Framework;
+using RegressionGames.StateRecorder.BotSegments.Models;
+using RegressionGames.TestFramework;
+using UnityEngine;
+using UnityEngine.UI;
+using Object = UnityEngine.Object;
+
+namespace RegressionGames.Tests.RGOverlay
+{
+    [TestFixture]
+    public class RGSegmentEntryTests
+    {
+        private GameObject _uat;
+
+        private RGSegmentEntry entry;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _uat = new GameObject();
+            entry = _uat.AddComponent<RGSegmentEntry>();
+            entry.segmentName = "TestSegment";
+            entry.description = "TestDescription";
+            entry.path = "test/path";
+            entry.type = BotSequenceEntryType.Segment;
+            entry.nameComponent = RGTestUtils.CreateTMProPlaceholder();
+            entry.descriptionComponent = RGTestUtils.CreateTMProPlaceholder();
+            entry.playButton = _uat.AddComponent<Button>();
+            
+            // create tooltip child
+            var tooltip = new GameObject();
+            tooltip.AddComponent<RGTooltip>();
+            tooltip.transform.SetParent(entry.transform);
+
+            // create segment list indicator w/Image
+            var segmentListIndicator = new GameObject();

```

Comment:
all uses of `new GameObject()` in these unit tests must follow the pattern...

```csharp
new GameObject(){
                transform =
                {
                    parent = parentObject.transform
                }
            };
```

where `parentObject` is something in the transform hierarchy of `_uat`.   


Otherwise, destroying `_uat` DOES NOT cleanup the other game objects  and leads to unit test interference between tests

Reply:
ah I'm glad you noticed this. I've updated my added tests